### PR TITLE
website

### PR DESCRIPTION
**Pull request by samuel-halim95**
First version of website.

### CODE REVIEW

**Comment by shankari** [comment]
ok to test

**Comment by shankari** [comment]
@samuel-halim95, I just pulled this into a private branch and tried to load the main page, and it still loaded the old index.html. I thought you were going to move that elsewhere so that the page that you designed is the main page.

**Comment by samuel-halim95** [comment]
Hi Shankari,

I’m sorry, but I don’t know how to do that. How do I change the main page into my home.html instead of index.html? Thank you.

Samuel Halim

**Comment by shankari** [comment]
You have two options:

- change your page to index.html and change the existing index.html to explore.html or something.
- change the code in `emission/net/api/cfc_webapp.py` right at the top where it returns `index.html` for the path `/`

**Comment by samuel-halim95** [comment]
allright fixed!

**Comment by shankari** [comment]
Did you use "git mv" to move the files around? Make sure to do that next
time so that history is preserved.

Also, did you try running this through bottle as opposed to loading it from
your file system? There are subtle changes between the two and you need to
check in the final deployment environment.

Look at the readme for details

**Comment by samuel-halim95** [comment]
I tried to run the server using: ./e-mission-py.bash emission/net/api/cfc_webapp.py.
but I got an error: Traceback (most recent call last):
  File "emission/net/api/cfc_webapp.py", line 27, in <module>
    import bson.json_util
ImportError: No module named json_util

**Comment by shankari** [comment]
Did you follow all the installation instructions including installing
anaconda and the modules in requirements.txt?

**Comment by samuel-halim95** [comment]
I just installed the modules in requirements.txt but it returns this error:
Exception:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/basecommand.py", line 209, in main
    status = self.run(options, args)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_set.py", line 726, in install
    requirement.uninstall(auto_confirm=True)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_install.py", line 746, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 303, in move
    os.unlink(src)
OSError: [Errno 13] Permission denied: '/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/dateutil/__init__.py'

**Comment by shankari** [comment]
Did you install anaconda first? Ensure that you are using included pip?

**Comment by samuel-halim95** [comment]
yeah I did, but it still gives me an error.

**Comment by shankari** [comment]
The error that you described is the result of using the standard pip, which requires sudo, not anaconda pip. Can you post the output of the checks for python and pip where we see if they are the anaconda versions?

**Comment by samuel-halim95** [comment]
MacBook-Pro-3:e-mission-server samuelhalim$ which python
/Library/Frameworks/Python.framework/Versions/2.7/bin/python
MacBook-Pro-3:e-mission-server samuelhalim$ which pip
/Library/Frameworks/Python.framework/Versions/2.7/bin/pip

**Comment by shankari** [comment]
Right, those are the standard python versions. See how they don't have `anaconda` like the examples in the README?

Make sure you are using the versions from anaconda and all will be fine.

**Comment by samuel-halim95** [comment]
How do I change it to anaconda?

**Comment by shankari** [comment]
Hopefully you learnt this in your OS class :) You pass the full path to the anaconda distribution, or you ensure that it is in your $PATH.

**Comment by samuel-halim95** [comment]
I haven't taken OS class yet :( Sorry about that

**Comment by samuel-halim95** [comment]
Installed successfull! but still get this error:
File "emission/net/api/cfc_webapp.py", line 30, in <module>
    import modeshare, zipcode, distance, tripManager, \
  File "/Users/samuelhalim/e-mission-server/emission/net/api/tripManager.py", line 13, in <module>
    import emission.net.ext_service.moves.collect as collect
  File "/Users/samuelhalim/e-mission-server/emission/net/ext_service/moves/collect.py", line 9, in <module>
    import numpy as np
ImportError: No module named numpy

**Comment by shankari** [comment]
That is still not using the current python distribution. numpy is included in the anaconda distribution. 
`e-mission-py.bash` assumes that the default `python` is the anaconda one. If the default is different, you need to edit `e-mission-py.bash` to specify the full path.

Also, anaconda puts itself into the path by default. Did you open a new terminal after the install?

**Comment by samuel-halim95** [comment]
allright all set. I got an Error: 500 Internal server when I tried to run the localhost

**Comment by shankari** [comment]
so what is the server error - there should be something in the terminal where you have started the server..

**Comment by samuel-halim95** [comment]
I've started the server and I wanna test the connection to the server, but I got that error instead.

**Comment by shankari** [comment]
Salim, I would like more detail - you got the error where - in the browser or in the server. is there a backtrace in the server terminal?

**Comment by samuel-halim95** [comment]
this is what I got from the terminal:
START 2016-03-21 18:33:35.609838 GET /favicon.ico
END 2016-03-21 18:33:35.610107 GET /favicon.ico  0.000238180160522 
Traceback (most recent call last):
  File "/Users/samuelhalim/e-mission-server/emission/net/api/bottle.py", line 852, in _handle
    self.trigger_hook('after_request')
  File "/Users/samuelhalim/e-mission-server/emission/net/api/bottle.py", line 630, in trigger_hook
    return [hook(*args, **kwargs) for hook in self._hooks[__name][:]]
  File "emission/net/api/cfc_webapp.py", line 430, in after_request
    msTimeNow, duration)
  File "/Users/samuelhalim/e-mission-server/emission/net/api/stats.py", line 84, in storeServerEntry
    get_server_stats_db_backup().insert(currEntry)
  File "/Users/samuelhalim/e-mission-server/emission/core/get_database.py", line 102, in get_server_stats_db_backup
    current_db=MongoClient().Stage_database
  File "/Users/samuelhalim/anaconda/lib/python2.7/site-packages/pymongo/mongo_client.py", line 377, in __init__
    raise ConnectionFailure(str(e))
ConnectionFailure: [Errno 61] Connection refused

**Comment by samuel-halim95** [comment]
The one that I posted earlier was from the browser

**Comment by shankari** [comment]
this one indicates that you haven't installed mongo, which is in the README.
Please follow ALL the steps in the README, you can't skip some of them and hope that it works.

**Comment by samuel-halim95** [comment]
I already installed mongodb

**Comment by samuel-halim95** [comment]
I followed all steps already.

**Comment by shankari** [comment]
The error that you got 
```
File "/Users/samuelhalim/anaconda/lib/python2.7/site-packages/pymongo/mongo_client.py", line 377, in __init
raise ConnectionFailure(str(e))
ConnectionFailure: [Errno 61] Connection refused
```
indicates that mongod is not running.

Did you start it?

File: emission/net/api/cfc_webapp.py [hunk 1]
```diff
@@ -73,17 +73,17 @@
...
 # Should we have gone for something like django instead after all?
 # If this gets to be too much, we should definitely consider that
 @route("/<filename>")
 def doc(filename):
-  if filename != "privacy" and filename != "support" and filename != "about" and filename != "consent":
+  if filename != "privacy" and filename != "home" and filename != "about" and filename != "index":
     return HTTPError(404, "Don't try to hack me, you evil spammer")
   else:
-    return static_file("%s.html" % filename, "%s/docs/" % static_path)
+    return static_file("%s.html" % filename, "%s/server/" % static_path)
```
Comment (shankari) [inline emission/net/api/cfc_webapp.py]: Did you check to see whether the existing
http://localhost:8080/privacy
and
http://localhost:8080/consent
URLs still work after this change?

Those are linked from the app store pages, and the IRB consent document. So we should ensure that they keep working.

If you have all the assets routes below, do you still need these?

File: front/server/index.html [hunk 2]
```diff
@@ -49,7 +49,7 @@
             dateFormat: "yy-mm-dd",
             defaultDate: "+1w",
             changeMonth: true,
-            numberOfMonths: 3,
+            numberOfMonths: 1,
```
Comment (shankari) [inline front/server/index.html]: was this an intentional change?

**Comment by samuel-halim95** [comment]
ummmm. I think it was a mistake. I didn't have any intention to change it. Sorry

**Comment by shankari** [comment]
what about the privacy and consent documents?

**Comment by samuel-halim95** [comment]
oh yeah I didn't check that and I forgot that we still need consent and privacy for that. sorry again

**Comment by shankari** [comment]
so can you check and push a fix?

**Comment by samuel-halim95** [comment]
Hi Shankari. I think I haven't pushed the newest one. I think there might still be some bugs on the latest one. I'll post that later in the night. Sorry for the delay.

**Comment by shankari** [comment]
Hi Samuel,

I couldn't wait any longer. We need to get this website out, and it's been a month since there has been no real update on it. I spent significant time last night and today getting everything to work properly, including moving to angular instead of jquery.

https://github.com/e-mission/e-mission-server/pull/275

I'll be deploying on the real server soon...

**Comment by samuel-halim95** [comment]
I think I just pushed it

**Pull request merged**